src/app/services/wallet.service.ts

```diff
@@ -64,16 +65,31 @@ export class WalletService {
       const minRequiredOutputs =  this.getMinRequiredOutputs(amount, outputs);
       const totalCoins = parseInt((minRequiredOutputs.reduce((count, output) =>
         count + output.coins, 0) * 1000000) + '', 10);
-      const totalHours = minRequiredOutputs.reduce((count, output) => count + output.hours, 0);
+
+      if (totalCoins < amount * 1000000) {
+        throw new Error('Not enough available SKY Hours to perform transaction!');
+      }
+
+      const totalHours = parseInt((minRequiredOutputs.reduce((count, output) =>
+        count + output.hours, 0)) + '', 10);
       const changeCoins = totalCoins - amount * 1000000;
-      const changeHours = Math.floor(totalHours * this.allocationRatio);
-      const txOutputs: TransactionOutput[] = [{ address: address, coins: amount * 1000000, hours: changeHours }];
+      let changeHours = parseInt((totalHours * this.allocationRatio) + '', 10);
+
+      const txOutputs: TransactionOutput[] = [];
       const txInputs: TransactionInput[] = [];
 
       if (changeCoins > 0) {
-        txOutputs.push({ address: wallet.addresses[0].address, coins: changeCoins, hours: changeHours });
+        txOutputs.push({
+          address: wallet.addresses[0].address,
+          coins: changeCoins,
+          hours: totalHours * this.unburnedHouarsRatio - changeHours
+        });
+      } else {
+        changeHours = parseInt((totalHours * this.unburnedHouarsRatio) + '', 10);
       }
 
+      txOutputs.push({ address: address, coins: amount * 1000000, hours: changeHours });
```

Comment:
The result of the multiplication must be filtered with `parseInt`, to prevent errors.